artifactory/commands/transferfiles/manager.go

```diff
@@ -125,6 +119,43 @@ func (ftm *transferManager) doTransfer(isProducerConsumer bool, transferAction t
 	return returnedError
 }
 
+type PollingTasksManager struct {
+	// Done channel notifies the polling go routines that no more tasks are expected.
+	doneChannel chan bool
+	// Error returned by the polling go routine
+	pollingErr error
+}
+
+// Runs 2 go routines :
+// 1. Check number of threads
+// 2. Poll uploaded chunks
+func (ptm *PollingTasksManager) start(runWaitGroup *sync.WaitGroup, producerConsumer parallel.Runner, uploadTokensChan chan string, srcUpService *srcUserPluginService, errorsChannelMng *ErrorsChannelMng) {
+	// Update threads by polling on the settings file.
+	runWaitGroup.Add(1)
+	go func() {
+		defer runWaitGroup.Done()
+		periodicallyUpdateThreads(producerConsumer, ptm.doneChannel)
+	}()
+
+	// Check status of uploaded chunks.
+	runWaitGroup.Add(1)
+	go func() {
+		defer runWaitGroup.Done()
+		ptm.pollingErr = pollUploads(srcUpService, uploadTokensChan, ptm.doneChannel, errorsChannelMng)
+	}()
+}
+
+func (ptm *PollingTasksManager) stop() {
+	// Notify the other go routines that work is done.
+	ptm.doneChannel <- true
```

Comment:
We can add to the struct an int member variable named `totalRunningGoRoutines`. The variable will be incremented by 1 before each of the two go routines as follows - 
```
	runWaitGroup.Add(1)
        ptm.totalRunningGoRoutines++
	go func() {
		defer runWaitGroup.Done()
		periodicallyUpdateThreads(producerConsumer, ptm.doneChannel)
	}()

	// Check status of uploaded chunks.
	runWaitGroup.Add(1)
        ptm.totalRunningGoRoutines++
	go func() {
		defer runWaitGroup.Done()
		ptm.pollingErr = pollUploads(srcUpService, uploadTokensChan, ptm.doneChannel, errorsChannelMng)
	}()
```

and then, instead of sending 2 close message `stop()` we can use the new `totalRunningGoRoutines` to determine how many messages to send. This way, when we add more polling go routines, we wouldn't need to remember to add another close message.

In addition, notice that you're holding the number 2 here as well - 
```
func newPollingTasksManager() PollingTasksManager {
	// Channel's size is 2, because there are exactly two go routines that need to be signaled that they should stop by it.
	return PollingTasksManager{doneChannel: make(chan bool, 2)}
}
```
The number 2 here can also be used as a struct member. You can call it `totaloGoRoutines`.Now to avoid developers forgetting to update this number, you can compare `totaloRunningGoRoutines` to `totaloGoRoutines` after all go routines are created, and return an error with a clear error message if the number aren't identical.